woocommerce/woocommerce-gutenberg-products-block
# Refactor Checkout Inner Block Registration and Render frontend forced blocks

This PR takes care of forced block rendering on the frontend. It does so by looking up forced blocks for an area, and appending it if there are no existing siblings with the same block name.

To achieve this, the registration system has been updated to accept `force` and store `block`, `component`, and `force`. These values are then looked up as needed.

This also makes it possible to register a checkout block without defining it on the PHP side. To do this, I used the`render_block` filter to append `data-block-name`. This applies to all blocks however, not just our blocks. I don't anticipate this causing any issues as it's just an attribute.

Closes #4574

### How to test the changes in this Pull Request:

Forced blocks

1. Edit the sample block and make it non forced 
2. Create a checkout block and save it with no sample blocks
3. Make sample block forced again
4. View frontend and confirm sample block is appended to the layout

Block registration

1. Confirm checkout i2 renders
2. Confirm single product block still renders, and can be updated/persisted without regression
3. Confirm all products block still renders, and can be updated/persisted without regression
